src/ui/popupmenu/lazy_loader.rs

```diff
@@ -67,7 +67,7 @@ impl LazyLoader {
         let mut state = self.state.borrow_mut();
         state.clear();
 
-        state.items_to_load = items;
+        state.items_to_load = items.clone();
```

Comment:
Calculate the `show_kind` value here, and if needed, store that to the `State` struct.